actionpack/lib/action_dispatch/middleware/static.rb

```diff
@@ -13,15 +13,13 @@ module ActionDispatch
   # located at `public/assets/application.js` if the file exists. If the file
   # does not exist, a 404 "File not Found" response will be returned.
   class FileHandler
-    def initialize(root, cache_control, index = 'index')
+    def initialize(root, index = 'index', headers: )
```

Comment:
Great.

This PR breaks the interface for other services/gems that hook into it, for instance the Heroku-deflator gem [here](https://github.com/romanbsd/heroku-deflater/blob/master/lib/heroku-deflater/serve_zipped_assets.rb#L15) but seems unavoidable for adding this feature.